### ISSUE TITLE
Optimizer Callback able to store gradient info for tensorboard

### ISSUE THREAD

**Issue by ogvalt** [catalyst-team/catalyst#576]
**Is your feature request related to a problem? Please describe.**
I was unable to harvest any gradient info in order to analyze my model training via tensorboard. I've tried to create a callback with CallbackOrder same as Optimizer, but it does not guarantee execution order.

**Describe the solution you'd like**
I would like to improve basic Optimizer Callback to be able to store gradient information for further analysis in tensorboard

**Describe alternatives you've considered**
The only alternative to achive this is to write everything in pytorch

**Additional context**
Here is an illustration of what I'd like to see as a result of enhancement: https://github.com/yunjey/pytorch-tutorial/tree/master/tutorials/04-utils/tensorboard

**Comment by TezRomacH** [comment]
Hi!
Given that `CallbackOrder` is just an integer, you can set it as `CallbackOrder.Optimizer - 1`

**Comment by ogvalt** [comment]
@TezRomacH The issue isn't in Callback Order, but in Optimizer CallBack itself. After self.grad_step it calls model.zero_grad() and all my grads become zeros and makes me unable to get correct grads. 
One solution that probably should work is to calculate grads on_batch begin without actually updating the model, but I see this as overhead.

**Comment by TezRomacH** [comment]
Alright, I see.

I'll think what we can do :)

**Comment by ogvalt** [comment]
@TezRomacH 
I thought a little and come to a solution. The simplest way is to add grads info to runner state and copy them before zero_grads in on_batch_end method call of Optimizer callback

**Comment by TezRomacH** [comment]
done in #581
